### PR TITLE
Add initial dependabot config

### PR DESCRIPTION
**Pull request by stone-z**

### CODE REVIEW

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 1
+update_configs:
+  - package_manager: "go:modules"
+    directory: "/"
+    update_schedule: "daily"
+    ignored_updates:
+      - match:
+          # Ignore k8s.io dependencies
+          dependency_name: "k8s.io/*"
```
Comment (tomahawk28) [inline .dependabot/config.yml]: perhaps can you add a newline?

Comment (stone-z) [reply .dependabot/config.yml]: :+1:

File: .dependabot/config.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+version: 1
+update_configs:
+  - package_manager: "go:modules"
+    directory: "/"
+    update_schedule: "daily"
+    ignored_updates:
```
Comment (tomahawk28) [inline .dependabot/config.yml]: Sorry, I think it's better to have `allowed_updates` with `github.com/giantswarm/*`. but we can change after this.

Comment (stone-z) [reply .dependabot/config.yml]: IMO external updates are as important if not more important than internal ones in general. Of course you can tune this, but external components introduce vulnerabilities and these should be monitored

Comment (rossf7) [reply .dependabot/config.yml]: Yes happy to try it with just `k8s.io` ignored. We can iterate if we need to add more later.

**Pull request merged**